### ISSUE TITLE
 Split bill – Receipt missing details when create a Split bill with 2 new users

### ISSUE THREAD

**Issue by lanitochka17** [Expensify/App#35925]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

**Version Number:** 1.4.37-2
**Reproducible in staging?:** Y
**Reproducible in production?:** Y
**If this was caught during regression testing, add the test name, ID and link from TestRail:** https://expensify.testrail.io/index.php?/tests/view/4282722
**Email or phone of affected tester (no customers):** applausetester+jp_e_category@applause.expensifail.com
**Logs:** https://stackoverflow.com/c/expensify/questions/4856
**Expensify/Expensify Issue URL:**
**Issue reported by:** Applause - Internal Team
**Slack conversation:**

## Action Performed:
1. Go to https://staging.new.expensify.com/
2. Login
3. Click on the FAB button > Request money
4. On the scan option, upload a digital receipt that will not fail the scan
5. On the participant selection list, select 2 other participants that you don't have a group created already
6. Continue to the final review page
7. Enter a description and click save button
8. Wait when Smartscan is complete
9. Click on the split preview

## Expected Result:
The amount and merchant fields are updated

## Actual Result:
Red dot and Receipt missing details message appears

## Workaround:
Unknown

## Platforms:

Which of our officially supported platforms is this issue occurring on?
- [ ] Android: Native
- [ ] Android: mWeb Chrome
- [ ] iOS: Native
- [ ] iOS: mWeb Safari
- [ ] MacOS: Chrome / Safari
- [ ] MacOS: Desktop

## Screenshots/Videos

Add any screenshot/video evidence

</details> 

https://github.com/Expensify/App/assets/78819774/0b629182-d25d-4f80-832f-e3f54519188e

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

**Comment by JordanLevy19** [comment]
**Proposal**

**Please re-state the problem that we are trying to solve in this issue.**
The problem arises from the receipt scan not recognizing or correctly assigning the "amount" from a receipt image uploaded for splitting expenses, leading to transactions displaying a "Receipt missing details" message.

**What is the root cause of that problem?**
The root cause of the problem appears to be a bug within the SmartScan's data extraction process. Specifically, identifying and parsing the amount from the uploaded receipt images.

**What changes do you think we should make in order to solve the problem?**

**Refine TransactionUtils.ts:**

Update the hasMissingSmartscanFields function to more accurately identify missing or incorrectly scanned data fields. This function checks if the transaction has missing required fields, which directly impacts the error state in the UI.

`function hasMissingSmartscanFields(transaction) {
    return transaction && hasReceipt(transaction) && !isDistanceRequest(transaction) && !isReceiptBeingScanned(transaction) && areRequiredFieldsEmpty(transaction);
}`

**Provide Manual Amount Entry as Fallback in MoneyRequestPreview:**

In the case of issues with the amount data, allow users to manually enter or correct the amount directly within the MoneyRequestPreview component. This could be implemented by introducing an input field 

`if (hasMissingSmartscanFields(transaction)) {
    return (
        <InputField
            label={translate('iou.enterAmountManually')}
            defaultValue={displayAmount}
            onChangeText={(value) => updateTransactionAmount(transactionID, value)}
        />
    );
}`

**Comment by akinwale** [comment]
Still looking for proposals here.

**Comment by strepanier03** [comment]
All good Melvin, waiting on proposals or urgency to move forward.

**Comment by strepanier03** [comment]
Proposals needed.

**Comment by strepanier03** [comment]
Waiting for proposals.

**Comment by strepanier03** [comment]
Tied to a VIP and waiting for proposals.

**Comment by strepanier03** [comment]
@akinwale - Somehow I missed this [proposal](https://github.com/Expensify/App/issues/35925#issuecomment-1934552696), did you review it by any chance?

**Comment by akinwale** [comment]
@strepanier03 Yes. The proposal does not solve the issue. Still looking for proposals.

**Comment by strepanier03** [comment]
Thank you for making sure @akinwale, I appreciate it 🙌

**Comment by strepanier03** [comment]
Awaiting ownership for a solution.

**Comment by strepanier03** [comment]
[Discussing](https://expensify.slack.com/archives/C01SKUP7QR0/p1710211827720109) whether or not to ask an expert agency to take on.

**Comment by strepanier03** [comment]
Since Melvin turned this internal, we can ask an agency if they want to work on it. Doing that now.

**Comment by strepanier03** [comment]
Posted [here](https://expensify.slack.com/archives/C03UK30EA1Z/p1710521639552309) in Callstack. If no one picks  it up, I'll post in SWM as well.

**Comment by Swor71** [comment]
hey, I'm Marcin from Callstack - expert contributor group - I'd like to help with this issue

**Comment by strepanier03** [comment]
I always make a fresh receipt, but I don't have a copy of it any longer I just toss it. I legit just type in something like the below and then screenshot it, then upload the screenshot as a receipt because it has all the information needed to smartscan successfully. 


Merchant: Walmart
Amount: $10
Date: 2024-03-19

@Swor71 are you unable to repro?

**Comment by Swor71** [comment]
hey @strepanier03, I asked as the invoice/receipt in the vid attached to this issue actually shows a bit more.. complex / regular / legit looking doc so I just wanted to make sure I'm testing it with correct file in order to eliminate a possible variable here, but if you're sure a screenshot with just those three lines is treated the same by smartscan then it's fine, thanks for that info 👍🏻

One the other note, I'm looking into this in detail but from the initial look, after uploading the file it says 'scan in progress' so I'd imagine it cannot populate those fields until that's finished? The other thing is that it seems to be stuck on that part, which might have to do with the smartscan itself, I'll see what I can find.

**Comment by Swor71** [comment]
hey, just as a quick update - I've had trouble reproducing the steps mentioned in the description given the method posted in the comments above (screenshot with just three 'fields' for smartscan) as it would just get stuck on the `scan in progress` when the steps and video evidence show/talk about smartscan actually finishing the job but the details would be missing.
This led me to believe that it would need a more regular looking document but that didn't work either. 
Test step nr 5, which talks about adding `2 other participants that you don't have a group created already` got me thinking and I tested the opposite - ie first create a chat and request money from there, this has actually produced a result of smartscan finishing, although it took well over 4 min to complete and populate the fields. 
I am currently looking into this piece and what is different between the two flows. Apologies for this taking a bit longer than (maybe) expected, there's quite a number of variables at work here.

**Comment by strepanier03** [comment]
No worries @Swor7, thank you for continuing to work on this 🙌

**Comment by mvtglobally** [comment]
Issue not reproducible during KI retests. (First week)

**Comment by Swor71** [comment]
@mvtglobally hey again, it would be really helpful if you could have a look at the questions above, thanks :)

**Comment by mvtglobally** [comment]
@Swor71 
QA team manually validates selected KIs in staging environment on the most recent build. 
There are cases when issue may be fixed or limited to particular accounts or devices where we may not be able to reproduce while another Contributor or Expensify team member can still see the issue on their side.

**Comment by strepanier03** [comment]
I've been OoO since last week, catching up on this now.

**Comment by strepanier03** [comment]
Hmmm, I've had a valid receipt scanning for an hour, not sure what's breaking down in the process but I'll circle back later after trying a new receipt.

**Comment by strepanier03** [comment]
> I've had trouble reproducing the steps mentioned in the description given the method posted in the comments above (screenshot with just three 'fields' for smartscan) as it would just get stuck on the scan in progress when the steps and video evidence show/talk about smartscan actually finishing the job but the details would be missing.
> This led me to believe that it would need a more regular looking document but that didn't work either.

Looks like I'm running into the same thing as @Swor71. 

I'm going to test the opposite as well and update here.

**Comment by strepanier03** [comment]
Hmmm, so after some additional testing I can confirm I get the same unending scan issue following the OP steps.  If the chat already exists, then the SS finishes and things are fine. This seems like a completely separate issue though.

I'm going to raise this for some internal discussion.

**Comment by Swor71** [comment]
hey, yea just to double confirm, the same keeps happening to me, tested again today. If there is a chat created before, the receipt gets scanned, although as you can see in the screenshot I had the dollar sign there but it set PLN as the currency, not sure if that's a bug or just a limitation of the scanner feature.

<img width="527" alt="image" src="https://github.com/Expensify/App/assets/32240132/2103a494-34e1-4371-be72-3c38a6685f41">

Also, to observe this issue happening you can create a money request with a random email address (doesn't have to be two ppl), which suggests to me it might have something to do with the creation of optimistic, placeholder, user - something I've previously seen to cause problems when assigning tasks to new users, maybe that helps.

This, however seems like a separate issue altogether, thanks for raising this internally @strepanier03. 

Just fyi I have been reassigned to a different project, currently awaiting the move so unfortunately I'm not sure how much I would be able to participate going further.

**Comment by strepanier03** [comment]
Thanks for the heads up @Swor71!

I expect to have an update next week.

**Comment by strepanier03** [comment]
No update yet, haven't been able to prioritize this yet.

**Comment by strepanier03** [comment]
This is on me, still working on it.

**Comment by strepanier03** [comment]
Nothing yet, sorry all.

**Comment by strepanier03** [comment]
Retesting today after releases have happened.

**Comment by strepanier03** [comment]
Using the following receipt image, I was able to test this successfully and did not reproduce the issue. 

The SmartScan took some time but it finished and there were no missing receipt details.

**Comment by MelvinBot** [comment]
This has been labelled "Needs Reproduction". Follow the steps here: https://stackoverflowteams.com/c/expensify/questions/16989
